clcache.py

```diff
@@ -1384,10 +1384,7 @@ def processDirect(cache, outputFile, compiler, cmdLine, sourceFile):
             listOfHeaderHashes = []
             for fileName in manifest.includeFiles:
                 fileHash = getFileHash(expandBasedirPlaceholder(fileName, baseDir))
-                if fileHash is not None:
-                    # May be if source does not use this header anymore (e.g. if that
-                    # header was included through some other header, which now changed).
-                    listOfHeaderHashes.append(fileHash)
+                listOfHeaderHashes.append(fileHash)
```

Comment:
I think this change is correct in itself, but it's based on a false premise: I thought it was suspicious that there's even a comment explaining why `None` may occur, so I checked the commit history and I think I did not port away from `getRelFileHash` correctly in 15eb86fe.

The function not only expanded the `CLCACHE_BASEDIR` placeholder, it also returned `None` if the given file does not exist. Thinking about it, I guess it's plausible: header files which existed in the past may no longer exist, so after expanding the placeholder, we should probably handle the case that the file does not exist. Something like

```python
try:
    fileHash = getFileHash(expandBasedirPlaceholder(fileName, baseDir))
    listOfHeaderHashes.append(fileHash)
except OSError:
    pass
```

...or so.